src/config_tools/config_tools.h

```diff
@@ -132,25 +134,55 @@ void ValidateIOConfig(TritonJson::Value &io_object, const IOConfig &io_config);
 
 
 /**
- * @brief Validates inputs array against provided config values
- * auto outputs auto-configured array to new_ios.
+ * @brief Validates inputs array against provided config values.
+ * Outputs auto-configured array to new_inputs.
+ *
+ * Auto-filled config keeps the order of inputs given in the original config file.
+ * If an input was not named in the original config, it is appended at the end.
+ * Inputs that do not appear in the original config are ordered lexicographically (pipeline order).
  */
 void AutofillInputsConfig(TritonJson::Value &inputs, const std::vector<IOConfig> &in_configs,
                           TritonJson::Value &new_inputs);
 
 
 /**
  * @brief Validates outputs array against provided config values
- * auto outputs auto-configured array to new_ios.
+ * Outputs auto-configured array to new_outputs.
+ *
+ * It's assumed that outputs in the config file have the same order as in the pipeline.
+ * If the config gives an output a name, it overrides the name coming from the pipeline.
  */
 void AutofillOutputsConfig(TritonJson::Value &outputs, const std::vector<IOConfig> &out_configs,
                            TritonJson::Value &new_outputs);
 
 
 /**
- * @brief Validates inputs or outputs array against provided config values.
+ * @brief Validate outputs array against provided config values
+ *
+ * Names of the outputs in the config file may not match names of the outputs in the pipeline.
```

Comment:
```suggestion
 * @brief Validate outputs array against provided config values.
 *
 * Names of the outputs in the config file do not need to match names of the outputs in the pipeline.
```
I believe `may not` is more like `they are not allowed`